### PR TITLE
Wr/add missing local components

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?

adds new remote component to `deploy report` output

### What issues does this PR fix or reference?
#https://github.com/forcedotcom/cli/issues/2603, @W-14675377@

### CODE REVIEW

File: src/formatters/deployResultFormatter.ts [hunk 1]
```diff
@@ -285,6 +286,26 @@ export class DeployResultFormatter extends TestResultsFormatter implements Forma
...
+      .map((s) =>
+        successes.push({
+          fullName: s.fullName,
+          // @ts-expect-error getState can return 'failed' which isn't applicable to FileSuccess
+          state: getState(s),
+          type: s.componentType ?? '',
+          filePath: 'Not found in project',
+        })
+      );
+
```
Comment (shetzel) [inline src/formatters/deployResultFormatter.ts:308]: IMO, SDR should do this.  Other clients may have the same problem and need this code.  I think SDR should report everything that happened, using the `ComponentSet` when possible to provide accurate file paths, but if not then use the file path from the response.  That means updating `MetadataApiDeploy.buildFileResponsesFromComponentSet()`.  @mshanemc - what are your thoughts?

**Review by shetzel — CHANGES_REQUESTED** [review]

File: src/formatters/deployResultFormatter.ts (hunk 1 above)
Comment (WillieRuemmele) [reply src/formatters/deployResultFormatter.ts:308]: I tried [that](https://github.com/forcedotcom/source-deploy-retrieve/pull/1273), but NUTs had errors, maybe there's a different method, or a new one, but I don't think `getFileResponses` is it

Comment (mshanemc) [reply src/formatters/deployResultFormatter.ts:308]: I like your SDR solution better (made 1 comment there).

I think the NUT failures look to all be `source-testkit` related, here: 
https://github.com/salesforcecli/source-testkit/blob/b5801a5ab89c044e94fd1ce14547ffe3dba9bf9f/src/assertions.ts#L120-L121

So maybe explore why that's hitting missing files?

Comment (WillieRuemmele) [reply src/formatters/deployResultFormatter.ts:308]: That would potentially break other consumers still

**Pull request closed without merging**